Compile errors after applying patches

I'm using `waf` to apply a collection of patches to `backstage` prior to compiling it. The patches are all successfully applied but the build process has two compile errors.

## Expected Behavior

`backstage` should compile (and run) after the patches are applied.

## Current Behavior

I get the following two errors during the build process:

```
plugins/auth-backend/src/lib/oauth/helpers.ts:39:5 - error TS2345: Argument of type 'Dictionary<string | undefined>' is not assignable to parameter of type 'string | Record<string, string> | string[][] | URLSearchParams | undefined'.
  Type 'Dictionary<string | undefined>' is not assignable to type 'string'.

39     pickBy(state, value => value !== undefined),
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

plugins/catalog-import/src/components/StepPrepareSelectLocations/StepPrepareSelectLocations.tsx:55:34 - error TS7006: Parameter 'l' implicitly has an 'any' type.

55     prepareResult?.locations.map(l => l.target) || [],
                                    ~


Found 2 errors.

error Command failed with exit code 1.
```

Note that neither of these files are patched!


## Context

I'm currently unable to build my `backstage` application.

## Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->

- NodeJS Version (v14): 14.17.5
- Operating System and Version (e.g. Ubuntu 14.04): MacOS Catalina

--- a/plugins/auth-backend/src/lib/oauth/helpers.ts
+++ b/plugins/auth-backend/src/lib/oauth/helpers.ts
@@ -36,7 +36,7 @@ export const readState = (stateString: string): OAuthState => {
 
 export const encodeState = (state: OAuthState): string => {
   const stateString = new URLSearchParams(
-    pickBy(state, value => value !== undefined),
+    pickBy<string>(state, value => value !== undefined),
   ).toString();
 
   return Buffer.from(stateString, 'utf-8').toString('hex');

